File: Java/src/Plocha.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Plocha {

    public List<ArrayList<String>> map = new ArrayList<ArrayList<String>>();
    public Node middle;

    public Plocha (int which){
        if (which == 1){
            map.add(new ArrayList<String>(Arrays.asList("T"," "," ","t"," "," "," ","T"," "," "," ","d"," "," ","T")));
            map.add(new ArrayList<String>(Arrays.asList(" "," ","d"," "," "," ","d"," ","d"," "," "," ","t"," "," ")));
            map.add(new ArrayList<String>(Arrays.asList(" ","t"," "," "," ","D"," "," "," "," "," "," "," ","d"," ")));
            map.add(new ArrayList<String>(Arrays.asList("d"," "," "," ","D"," "," ","d"," "," ","D"," "," "," ","d")));
            map.add(new ArrayList<String>(Arrays.asList(" "," "," ","D"," "," "," "," "," "," "," ","D"," "," "," ")));
            map.add(new ArrayList<String>(Arrays.asList(" "," "," "," "," ","t"," "," "," ","t"," "," ","D"," "," ")));
            map.add(new ArrayList<String>(Arrays.asList(" ","t"," "," "," "," ","d"," ","d"," "," "," "," ","d"," ")));
            map.add(new ArrayList<String>(Arrays.asList("T"," "," ","d"," "," "," ","S"," "," "," ","d"," "," ","T")));
            map.add(new ArrayList<String>(Arrays.asList(" ","d"," "," "," "," ","d"," ","d"," "," "," "," ","d"," ")));
            map.add(new ArrayList<String>(Arrays.asList(" "," ","D"," "," ","t"," "," "," ","t"," "," "," "," "," ")));
            map.add(new ArrayList<String>(Arrays.asList(" "," "," ","D"," "," "," "," "," "," "," ","D"," "," "," ")));
            map.add(new ArrayList<String>(Arrays.asList("t"," "," "," ","D"," "," ","d"," "," ","D"," "," "," ","d")));
            map.add(new ArrayList<String>(Arrays.asList(" ","d"," "," "," "," "," "," "," ","D"," "," "," ","t"," ")));
            map.add(new ArrayList<String>(Arrays.asList(" "," ","t"," "," "," ","d"," ","d"," "," "," ","d"," "," ")));
            map.add(new ArrayList<String>(Arrays.asList("T"," "," ","d"," "," "," ","T"," "," "," ","t"," "," ","T")));
        }

        //create_nodes();
    }

    public List<ArrayList<Node>> create_nodes(){
        List<ArrayList<Node>> out = new ArrayList<>();
        for (ArrayList<String> line : map){
            out.add(new ArrayList<Node>());
            ArrayList<Node> last_array = out.get(out.size()-1);
            boolean first_line = map.indexOf(line) == 0;
            for (String pom : line){
                Node node = new Node(pom);
                last_array.add(node);
                boolean left_line = line.indexOf(pom) == 0;

                if (pom.equals("|")) node.type = Node.NodeType.TRIPLE_WHOLE;
                else if (pom.equals("=")) node.type = Node.NodeType.DOUBLE_WHOLE;
                else if (pom.equals("T")) node.type = Node.NodeType.TRIPLE;
                else if (pom.equals("D")) node.type = Node.NodeType.DOUBLE;
                else if (pom.equals("S")){
                    node.type = Node.NodeType.START;
                    middle = node;
                }
                else node.type = Node.NodeType.REGULAR;


                if (!first_line && !left_line){
                    node.add_neightbour(out.get(out.indexOf(last_array) - 1).get(last_array.indexOf(node)),last_array.get(last_array.indexOf(node) - 1));
                }
                else if (!first_line){
                    node.add_neightbour(out.get(out.indexOf(last_array) - 1).get(last_array.indexOf(node)),"top");
                }
                else if (!left_line){
                    node.add_neightbour(last_array.get(last_array.indexOf(node) - 1),"left");
                }
            }
        }
        return out;
    }
    public boolean is_empty(){
        return "S".equals(middle.name);
    }
}

File: C#/ConsoleApp1/ConsoleApp1/Plocha.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace ClassLibrary1
{
    public class Plocha
    {
        public List<List<string>> map = new List<List<string>>();
        public Node middle;

    public Plocha (int which){
        if (which == 1){
            map.Add(new List<string> {"T"," "," ","t"," "," "," ","T"," "," "," ","d"," "," ","T"});
            map.Add(new List<string> {" "," ","d"," "," "," ","d"," ","d"," "," "," ","t"," "," "});
            map.Add(new List<string> {" ","t"," "," "," ","D"," "," "," "," "," "," "," ","d"," "});
            map.Add(new List<string> {"d"," "," "," ","D"," "," ","d"," "," ","D"," "," "," ","d"});
            map.Add(new List<string> {" "," "," ","D"," "," "," "," "," "," "," ","D"," "," "," "});
            map.Add(new List<string> {" "," "," "," "," ","t"," "," "," ","t"," "," ","D"," "," "});
            map.Add(new List<string> {" ","t"," "," "," "," ","d"," ","d"," "," "," "," ","d"," "});
            map.Add(new List<string> {"T"," "," ","d"," "," "," ","S"," "," "," ","d"," "," ","T"});
            map.Add(new List<string> {" ","d"," "," "," "," ","d"," ","d"," "," "," "," ","d"," "});
            map.Add(new List<string> {" "," ","D"," "," ","t"," "," "," ","t"," "," "," "," "," "});
            map.Add(new List<string> {" "," "," ","D"," "," "," "," "," "," "," ","D"," "," "," "});
            map.Add(new List<string> {"t"," "," "," ","D"," "," ","d"," "," ","D"," "," "," ","d"});
            map.Add(new List<string> {" ","d"," "," "," "," "," "," "," ","D"," "," "," ","t"," "});
            map.Add(new List<string> {" "," ","t"," "," "," ","d"," ","d"," "," "," ","d"," "," "});
            map.Add(new List<string> {"T"," "," ","d"," "," "," ","T"," "," "," ","t"," "," ","T"});
        }

        //create_nodes();
    }

    public List<List<Node>> create_nodes(){
        List<List<Node>> output = new List<List<Node>>();
        foreach (var line in map)
        {
            output.Add(new List<Node>());
            List<Node> lastArray = output[output.Count-1];
            bool firstLine = map.IndexOf(line) == 0;
            foreach (var pom in line)
            {
                Node node = new Node(pom);
                lastArray.Add(node);
                bool leftLine = line.IndexOf(pom) == 0;

                if (pom == "|") node.type = Node.NodeType.TRIPLE_WHOLE;
                else if (pom == "=") node.type = Node.NodeType.DOUBLE_WHOLE;
                else if (pom == "T") node.type = Node.NodeType.TRIPLE;
                else if (pom == "D") node.type = Node.NodeType.DOUBLE;
                else if (pom == "S"){
                    node.type = Node.NodeType.START;
                    middle = node;
                }
                else node.type = Node.NodeType.REGULAR;


                if (!firstLine && !leftLine){
                    node.add_neightbour(output[output.IndexOf(lastArray) - 1][lastArray.IndexOf(node)],lastArray[lastArray.IndexOf(node) - 1]);
                }
                else if (!firstLine){
                    node.add_neightbour(output[output.IndexOf(lastArray) - 1][lastArray.IndexOf(node)],"top");
                }
                else if (!leftLine){
                    node.add_neightbour(lastArray[lastArray.IndexOf(node) - 1],"left");
                }
            }
        }
        return output;
    }
    public bool is_empty(){
        return "S" == middle.name;
    }
    }
}